### PR TITLE
feat: add spreadsheet documentation

### PR DESCRIPTION
**Pull request by yuriy-fix**

### CODE REVIEW

**Review by jouni — COMMENTED** [review]
The main `spreadsheet.png` image looks like it’s still using the FontAwesome icons.

File: articles/ds/components/spreadsheet/index.asciidoc [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+
+Hyperlinks in cells can point to other worksheets in the current workbook, or to external URLs.
+Links must be added through the POI API.
+Spreadsheet provides default handling for hyperlink clicks, which can be overridden with a `customHyperlinkCellClickHandler`, which you assign `withsetHyperlinkCellClickHandler()`.
+
+== Popup Buttons in Cells
+You can add a popup button in a cell, clicking which opens a drop-down popup overlay, which can contain any Vaadin components.
+You can add a popup button with any of the `setPopup()` methods for different cell addressing forms.
+A popup button is a PopupButton object, which is a regular Vaadin component container, so you can add any components to it by `addComponent()`.
+You can create popup buttons for a row of cells in a cell range by defining a table, as described in link:https://vaadin.com/docs/v8/spreadsheet/spreadsheet-table#spreadsheet.table[Tables Within Spreadsheets].
```
Comment (jouni) [inline articles/ds/components/spreadsheet/index.asciidoc:193]: Are these popups supported yet? At least the link to Vaadin 8 docs needs to be removed.

File: articles/ds/components/spreadsheet/index.asciidoc [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+When a sheet is deleted, the index of the sheets with a higher index is decremented by one.
+When the active worksheet is deleted, the next one by index is set as active, or if there are none, the previous one.
+
+All operations on the spreadsheet content are done through the currently active worksheet.
+You can set an existing sheet as active with `setActiveSheetIndex()`.
+
+
+== Working with an Apache POI Workbook
+
+If you have an Apache POI workbook created otherwise, you can wrap it toSpreadsheet with the respective constructor.
```
Comment (dougjewsbury) [inline articles/ds/components/spreadsheet/index.asciidoc:68]: Can the meaning be made clearer?

File: articles/ds/components/spreadsheet/index.asciidoc [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+
+You can access the underlying workbook with `getWorkbook()`.
+However, if you make modifications to the workbook, you must allow the spreadsheet update itself by calling appropriate update methods for the modified element or elements.
```
Comment (dougjewsbury) [inline articles/ds/components/spreadsheet/index.asciidoc:71]:
```suggestion
However, if you make modifications to the workbook, you must allow the spreadsheet to update itself by calling appropriate update methods for the modified element or elements.
```

File: articles/ds/components/spreadsheet/index.asciidoc (hunk 2 above)
Comment (dougjewsbury) [inline articles/ds/components/spreadsheet/index.asciidoc:68]: Can the meaning of this sentence be made clearer?

File: articles/ds/components/spreadsheet/index.asciidoc (hunk 3 above)
Comment (dougjewsbury) [inline articles/ds/components/spreadsheet/index.asciidoc:71]:
```suggestion
However, if you make modifications to the workbook, you must allow the spreadsheet to update itself by calling appropriate update methods for the modified element or elements.
```

File: articles/ds/components/spreadsheet/index.asciidoc [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+You can control its visibility with `setSheetSelectionBarVisible()`.
+
+|Report mode
+|In the report mode, both the top and bottom bars are hidden. You can enable the mode with `setReportStyle()`.
+
+|===
+
+== Frozen Row and Column Panes
+
+You can define panes of rows and columns that are frozen in respect to scrolling.
```
Comment (dougjewsbury) [inline articles/ds/components/spreadsheet/index.asciidoc:101]:
```suggestion
You can define panes of rows and columns that are frozen when scrolling.
```

File: articles/ds/components/spreadsheet/index.asciidoc [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+----
+--
+
+== Cell Comments
+
+Cells may have comments that are indicated by ticks in the corner of the cells, and the comment is shown when mouse is hovered on the cells.
+`TheSpreadsheetDefaultActionHandler` described in <<Default Context Menu>> enables adding comments from the context menu.
+
+A new comment can be added through the POI API of a cell, with `addComment()`.
+For a detailed example for managing cell comments, we refer to the `InsertDeleteCellCommentAction` and `EditCellCommentAction` classes employed by the default action handler.
```
Comment (dougjewsbury) [inline articles/ds/components/spreadsheet/index.asciidoc:170]: Can the meaning of this sentence be made clearer?

File: articles/ds/components/spreadsheet/index.asciidoc [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+
+== Hyperlinks
+
+Hyperlinks in cells can point to other worksheets in the current workbook, or to external URLs.
+Links must be added through the POI API.
+Spreadsheet provides default handling for hyperlink clicks, which can be overridden with a `customHyperlinkCellClickHandler`, which you assign `withsetHyperlinkCellClickHandler()`.
+
+== Popup Buttons in Cells
+You can add a popup button in a cell, clicking which opens a drop-down popup overlay, which can contain any Vaadin components.
+You can add a popup button with any of the `setPopup()` methods for different cell addressing forms.
```
Comment (dougjewsbury) [inline articles/ds/components/spreadsheet/index.asciidoc:191]: Can it be made clearer what "different cell addressing forms" refers to?

File: articles/ds/components/index.asciidoc [hunk 7]
```diff
@@ -103,6 +103,12 @@ image::{components-path-prefix}select/select.png["", opts=inline, role="icon scr
 include::select/index.asciidoc[tag=description]
 <<{components-path-prefix}select#,See Select>>
 
+[.component-card]
+=== Spreadsheet
+image::{components-path-prefix}spreadsheet/spreadsheet.png["", opts=inline, role="icon screenshot"]
+include::spreadsheet/index.asciidoc[tag=description]
```
Comment (manolo) [inline articles/ds/components/index.asciidoc:109]: does the `spreadsheet/index.asciidoc` exist? at leas not in this branch

**Review by manolo — CHANGES_REQUESTED** [review]

**Comment by jouni** [comment]
Closing as stale (keeping the branch available for now).

**Pull request closed without merging**